test/executor/loader_test.cpp

```diff
@@ -128,8 +128,28 @@ TEST_F(LoaderTests, LoadingTest) {
   LaunchParallelTest(loader_threads_count, InsertTuple, data_table.get(),
                      testing_pool, tilegroup_count_per_loader);
 
-  auto expected_tile_group_count =
-      loader_threads_count * tilegroup_count_per_loader + 1;
+  auto expected_tile_group_count = 0;
+
+  int total_tuple_count = loader_threads_count * tilegroup_count_per_loader * TEST_TUPLES_PER_TILEGROUP;
```

Comment:
This is necessary if we want to support scalable tile group allocation. The calculation of number of allocated tile groups is complicated.